nucypher/blockchain/eth/currency.py

```diff
@@ -0,0 +1,197 @@
+        return int(self.to_nu_wei())
+
+    def __repr__(self) -> str:
+        r = f'{self.__symbol}(value={str(self.__value)})'
+        return r
+
+    def __str__(self) -> str:
+        return f'{str(self.to_tokens())} {self.__symbol}'
+
+
+class Stake:
+    """
+    A quantity of tokens, and staking time-frame for one stake for one miner.
+    """
+
+    __ID_LENGTH = 16
+
+    def __init__(self,
+                 owner_address: str,
+                 index: int,
+                 value: NU,
+                 start_period: int,
+                 end_period: int):
+
+        # Stake Info
+        self.owner_address = owner_address
+        self.index = index
+        self.value = value
+        self.start_period = start_period
+        self.end_period = end_period
+        self.duration = self.end_period - self.start_period
+
+        # Internals
+        self.start_datetime = datetime_at_period(period=start_period)
+        self.end_datetime = datetime_at_period(period=end_period)
+        self.duration_delta = self.end_datetime - self.start_datetime
+
+    def __repr__(self):
+        r = f'Stake(index={self.index}, value={self.value}, end_period={self.end_period})'
+        return r
+
+    def __eq__(self, other):
+        return bool(self.value == other.value)
+
+    @classmethod
+    def from_stake_info(cls, owner_address, index: int, stake_info: Tuple[int, int, int]):
+        """Reads staking values as they exist on the blockchain"""
+        start_period, end_period, value = stake_info
+        instance = cls(owner_address=owner_address,
+                       index=index,
+                       start_period=start_period,
+                       end_period=end_period,
+                       value=NU(value, 'NUWei'))
+        return instance
+
+    def to_stake_info(self) -> Tuple[int, int, int]:
+        """Returns a tuple representing the blockchain record of a stake"""
+        return self.start_period, self.end_period, int(self.value)
+
+    @property
+    def id(self) -> str:
+        """TODO: Unique staking ID, currently unused"""
+        digest = b''
+        digest += eth_utils.to_canonical_address(address=self.owner_address)
+        digest += str(self.index).encode()
+        digest += str(self.start_period).encode()
+        digest += str(self.end_period).encode()
+        digest += str(self.value).encode()
+        stake_id = sha256(digest).hex()[:16]
+        return stake_id[:self.__ID_LENGTH]
+
+    @property
+    def periods_remaining(self) -> int:
+        """Returns the number of periods remaining in the stake from now."""
+        current_period = datetime_to_period(datetime=maya.now())
+        return self.end_period - current_period
+
+    def time_remaining(self, slang: bool = False) -> Union[int, str]:
+        """Returns the time delta remaining in the stake from now."""
+        now = maya.now()
+        delta = self.end_datetime - now
+        if slang:
+            result = self.end_datetime.slang_date()
```

Comment:
```suggestion
            result = delta.slang_date()
```

Reply:
That will result in an `AttributeError`: <https://circleci.com/gh/nucypher/nucypher/21089?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link>.

I think it is correct the way it is now, see: <https://github.com/kennethreitz/maya>

Reply:
Ok I see. I got confused because this method returns either an interval of time (in seconds, when `slang=False`) or an absolute date (when `slang=True`).